ktesting/testinglogger_test.go

```diff
@@ -145,3 +153,63 @@ func (l *logToBuf) Log(args ...interface{}) {
 	}
 	l.Write([]byte("\n"))
 }
+
+func TestStop(t *testing.T) {
+	// This test is set up so that a subtest spawns a goroutine and that
+	// goroutine logs through ktesting *after* the subtest has
+	// completed. This is not supported by testing.T.Log and normally
+	// leads to:
+	//   panic: Log in goroutine after TestGoroutines/Sub has completed: INFO hello world
+	//
+	// It works with ktesting if (and only if) logging gets redirected to klog
+	// before returning from the test.
+
+	// Capture output for testing.
+	state := klog.CaptureState()
+	defer state.Restore()
+	var output bytes.Buffer
+	var fs flag.FlagSet
+	klog.InitFlags(&fs)
+	fs.Set("alsologtostderr", "false")
+	fs.Set("logtostderr", "false")
+	fs.Set("stderrthreshold", "FATAL")
+	fs.Set("one_output", "true")
+	klog.SetOutput(&output)
+
+	var line int
+	var wg1, wg2 sync.WaitGroup
+	wg1.Add(1)
+	wg2.Add(1)
+	t.Run("Sub", func(t *testing.T) {
+		logger, _ := ktesting.NewTestContext(t)
+		go func() {
+			defer wg2.Done()
+
+			// Wait for test to have returned.
+			wg1.Wait()
+
+			// This output must go to klog because the test has
+			// completed.
+			_, _, line, _ = runtime.Caller(0)
+			logger.Info("simple info message")
+			logger.Error(nil, "error message")
+			logger.WithName("me").WithValues("completed", true).Info("complex info message", "anotherValue", 1)
+		}()
+	})
+	// Allow goroutine above to proceed.
+	wg1.Done()
+
+	// Ensure that goroutine has completed.
+	wg2.Wait()
+
+	actual := output.String()
```

Comment:
Can we please also extend this test to show that these lines are not captured in the buffer?

Reply:
If we setup your `ktesting.NewTestContext(t)` to a global `logger logr.Logger` we can do the following to show how things work when someone writes to logger post test context. Would this be useful to have ?
```golang
	testingLogger, _ := logger.GetSink().(ktesting.Underlier)
	if len(testingLogger.GetBuffer().Data()) != 1 {
		t.Error("Expected this one to not catch logs from leaked goroutines")
	}
```

Reply:
Please mark this as resolved